lib/src/warnings.dart

```diff
@@ -14,19 +12,22 @@ import 'package:dartdoc/src/model/comment_referable.dart';
 import 'package:dartdoc/src/model/model.dart';
 import 'package:dartdoc/src/package_meta.dart';
 
+const _namePlaceholder = '@@name@@';
+
 abstract class PackageWarningOptionContext implements DartdocOptionContextBase {
   bool get allowNonLocalWarnings =>
       optionSet['allowNonLocalWarnings'].valueAt(context);
 
-  // allowWarningsInPackages, ignoreWarningsInPackages, errors, warnings, and ignore
-  // are only used indirectly via the synthetic packageWarningOptions option.
+  // allowWarningsInPackages, ignoreWarningsInPackages, errors, warnings, and
+  // ignore are only used indirectly via the synthetic packageWarningOptions
+  // option.
   PackageWarningOptions get packageWarningOptions =>
       optionSet['packageWarningOptions'].valueAt(context);
 
   bool get verboseWarnings => optionSet['verboseWarnings'].valueAt(context);
 }
 
-Future<List<DartdocOption<Object>>> createPackageWarningOptions(
+Future<List<DartdocOption<Object?>>> createPackageWarningOptions(
```

Comment:
You might be able to drop the Object parameter here, that has worked in other cases.

Reply:
All of the DartdocOption objects inside are DartdocOption<some nullable type> so it needed to be nullable.

Reply:
oh.  I guess I was suggesting to use an implicit dynamic here.   Is there a best practice?

Reply:
Oh I see. I avoid dynamic at all costs 😁 . The leading recommendation in that vein is probably Effective Dart's https://dart.dev/guides/language/effective-dart/design#avoid-using-dynamic-unless-you-want-to-disable-static-checking